### PR TITLE
migrate bot to typescript

### PR DESCRIPTION
**Pull request by traumschule**
See #23

### CODE REVIEW

**Comment by freakstatic** [comment]
I did a quick review and it seems very good!
Good job :)

**Comment by mochet** [comment]
This was approved by the council here: https://testnet.joystream.org/#/proposals/32

**Comment by bwhm** [comment]
@DzhideX 
Do you mind having a look at the code? I'm running some tests now, but it would be nice if you could have a closer look :)

File: community-contributions/joystreamtelegrambot/config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// website
+export const domain = "https://testnet.joystream.org";
+
+export const wsLocation = "ws://rome-rpc-endpoint.joystream.org:9944/";
+// websocket location
```
Comment (DzhideX) [inline community-contributions/joystreamtelegrambot/config.ts:5]: I don't know if this is something just on my side here but running the bot I'd get this: 

```
2020-10-30 12:14:38          API-WS: disconnected from ws://rome-rpc-endpoint.joystream.org:9944/ code: '1006' reason: 'connection failed'
```

Changing `ws` to `wss` seems to fix it.

This is nitpicking but maybe also move the comment above the declaration? :)

File: community-contributions/joystreamtelegrambot/src/bot.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+const opts: Options = parseArgs(process.argv.slice(2));
+const log = (msg: string): void | number => opts.verbose && console.log(msg);
+process.env.NTBA_FIX_319 ||
+  log("TL;DR: Set NTBA_FIX_319 to hide this warning.");
+
+const bot = new TelegramBot(token, { polling: true });
+
+const sendMessage = (msg: string) => {
+  try {
+    //bot.sendMessage(chatid, msg, { parse_mode: "HTML" });
```
Comment (DzhideX) [inline community-contributions/joystreamtelegrambot/src/bot.ts:24]: I assume this is not supposed to be commented out?

File: community-contributions/joystreamtelegrambot/src/lib/announcements.ts [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  proposalDetail
+} from "./getters";
+
+const sleep = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+const query = async (
+  test: string,
+  callback: () => Promise<any>
+): Promise<any> => {
+  let result: any = await callback();
```
Comment (DzhideX) [inline community-contributions/joystreamtelegrambot/src/lib/announcements.ts:20]: Result will implicitly figure out the type to be any here, no need to explicitly write it. 

There's more similar situations in the files.

File: community-contributions/joystreamtelegrambot/src/lib/getters.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+export const memberHandle = async (api: Api, id: number): Promise<string> => {
+  const membership: Membership = await api.query.members.membershipById(id);
+  return membership.handle.toJSON();
+};
+
+export const memberHandleByAccount = async (
+  api: Api,
+  account: string
+): Promise<string> => {
+  const id: number = await api.query.members.memberIdsByRootAccountId(account);
```
Comment (DzhideX) [inline community-contributions/joystreamtelegrambot/src/lib/getters.ts:32]: I think we can use `MemberId` here, unless there's a reason not to?

There are also some necessary changes on line 23, 103 and announcements.ts

**Review by DzhideX — CHANGES_REQUESTED** [review]
Hi @traumschule,

this is really good but I have some small remarks that should be fixed.The two changes in `bot.ts` and `config.ts` are vital whereas the rest are mostly cosmetic. 

After you've made all of the changes it would be nice if you could just delete all of the unused imports, variables, commented out logic as well.

Thank you for your contribution :100:

**Comment by bwhm** [comment]
As we didn't complete the review in time, this will be graded as fully implemented, ref [this](https://github.com/Joystream/community-repo/issues/23). 
Although there is no indication that the third success event (see below) is fixed, the full amount is paid.
```
The 'bug' associated with the finalization of Proposals are fixed 
```

I will merge it now, but we will perform a few small fixes.

**Pull request merged**